problem_2.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Q. Given an array of numbers of length N, find both the minimum and maximum.
+ * Follow up : Can you do it using less than 2 * (N - 2) comparison
+ * 
+ * Approach 1(Simple) - iterate the array and keep track of min and max elemets
+ * using two variables. For each number we make two comparision hence total
+ * number of comparision is 2n.
+ * 
+ * Time Complexity - O(n) 
+ * Space Complexity - O(1)
+ * 
+ * Approach 2(Divide and conquer approach) - Keep on dividing the array
+ * recursively until we have two or one element. Compare the elements and find
+ * min and max. Now like merge sort instead of combining the two left and right
+ * side of partition, we compare their min and max. 
```

Comment:
great and unique approach!!
you are thinking in the right direction!!
keep on doing this!!